src/internal/m365/collection/drive/collections.go

```diff
@@ -147,7 +147,11 @@ func deserializeMetadata(
 				// these cases. We can make the logic for deciding when to continue vs.
 				// when to fail less strict in the future if needed.
 				if err != nil {
-					return nil, nil, false, clues.Stack(err).WithClues(ictx)
+					errs.Fail(clues.Stack(err).WithClues(ictx))
+
+					breakLoop = true
+
+					break
```

Comment:
I initially did this as this was the only way to get `deserializeMetadata` to return something other than true for `canUsePreviousBackup`. But this feels like we should ideally not be having a hard fail on this. What we there was some network issue midway through writing the metadata file? Any thoughts @ashmrtn @ryanfkeepers

Reply:
we should be able to assume that either the metadata file is complete (written fully but possibly malformed if there was a bug in the code) or not there at all due to how kopia works. We're ignoring things like bit rot and bit flips for the moment, but those are much more involved to handle

is there a reason we can't continue to just return false here? Since we don't know what we don't know if we see an error (i.e. we don't know what the previous drives were) we should just fetch everything again. I don't expect we'll have multiple metadata file sets, but I think the conservative approach of assuming we can't trust anything if one thing is bad is ok. I'm honestly not sure how to reason about potential overlap between different meatadata file sets

Reply:
also, I think the end goal should be creating/returning tombstone collections in code in files like this rather than the caller of this code. I realize that's a much larger change though so it's not reasonable to expect it to be done in this PR

Reply:
> is there a reason we can't continue to just return false here?

don't see why not.  This is doing the same eventual effect, just via a longer path.  Returning here will skip the block on 167, but that's redundant anyway since we'll end up inside 187 regardless.

> But this feels like we should ideally not be having a hard fail on this.

Agreed. Seems reasonable.  Since we're using the return value on 190 anyway (empty maps, false, nil), why not replicate that return here?

> I think the end goal should be creating/returning tombstone collections in code in files like this rather than the caller of this code

100% agreed.  Two questions: 1/ should we make a tech debt ticket to track the refactor, or include it as a v2 in this change?  2/ The more we democratize collection creation across multiple owners, does that increase the possibility of producing conflicting collection sets?  Is this something we should have preventative checks at some higher level to avoid?

Reply:
>Two questions: 1/ should we make a tech debt ticket to track the refactor, or include it as a v2 in this change?

Tech debt may be better. It's going to end up touching all service collection generators since they all have it right now. While I'd like to see the work done, I don't want to bog down other things by requiring it be done with this work

>2/ The more we democratize collection creation across multiple owners, does that increase the possibility of producing conflicting collection sets? Is this something we should have preventative checks at some higher level to avoid?

There's some basic checks on this in kopia wrapper, for checking that there's not multiple collections referring to the same FullPath(). I don't recall if there's a check that there's not multiple collections referring to the same PreviousPath(), there may be. Are there other checks you think we should have?

Also note that the checks above are _runtime_ checks. We can't tell before hand if there'll be collisions in an easy manner. It's one of the things we kind of have to reason through when writing the code

Reply:
> Are there other checks you think we should have?

Not off the top of my head.  But if memory serves, those checks will fail out the backup, right?  Ideally, we'd be able to "fix" the state before failing out (not sure off the top of my head).  Getting into a state of consistent failure (or, perhaps worse, inconsisstent failure) due to this scares me a bit, since replication of that state might be very difficult.

Reply:
> also, I think the end goal should be creating/returning tombstone collections in code in files like this rather than the caller of this code. I realize that's a much larger change though so it's not reasonable to expect it to be done in this PR

Agreed, I am planning to pick it up next. Just wanted to do the minimum necessary to ensure correctness of groups backup first. Lemme create a [tech-debt ticket](https://github.com/alcionai/corso/issues/4405)